data-protocols/dsp/dsp-http-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/http/api/configuration/DspApiConfigurationExtension.java

```diff
@@ -141,6 +143,7 @@ private void registerTransformers() {
         dspApiTransformerRegistry.register(new JsonObjectFromDataAddressDspaceTransformer(jsonBuilderFactory, mapper));
         dspApiTransformerRegistry.register(new JsonObjectFromQuerySpecTransformer(jsonBuilderFactory));
         dspApiTransformerRegistry.register(new JsonObjectFromCriterionTransformer(jsonBuilderFactory, mapper));
+        dspApiTransformerRegistry.register(new JsonObjectFromSecretTransformer(jsonBuilderFactory));
```

Comment:
The registration of the secrets transformers has nothing to do with the DSP protocol, so its registration should **not** be done here, as that would introduce a dependency from DSP -> Secrets. Better to do this in the `ManagementApiConfigurationExtension`, or the `SecretsApiExtension`

Reply:
Agree and I would put it in the `SecretsApiExtension`